File: app/src/main/java/me/pwcong/jpstart/network/baidu/BaiDuTranslateApi.java
package me.pwcong.jpstart.network.baidu;

import me.pwcong.jpstart.mvp.bean.BaiduTranslateBean;
import retrofit2.http.GET;
import retrofit2.http.Query;
import rx.Observable;

/**
 * Created by Pwcong on 2016/10/1.
 */

public interface BaiduTranslateApi {

    String AUTO = "auto";
    String ZH = "zh";
    String JP = "jp";
    String EN = "en";

    String OK = "52000";
    String ERROR_TIMEOUT = "52001";
    String ERROR_SYSTEM = "52002";
    String ERROR_APPID = "52003";
    String ERROR_PARAMS = "54000";
    String ERROR_ILLEGAL_IP = "58000";
    String ERROR_SIGN = "54001";
    String ERROR_FREQUENT = "54003";
    String ERROR_TRANSLATE = "58001";
    String ERROR_NO_MONEY = "54004";
    String ERROR_LONG_FREQUENT = "54005";

    @GET("/api/trans/vip/translate")
    Observable<BaiduTranslateBean> request(@Query("q") String q, @Query("from") String from, @Query("to") String to,
            @Query("appid") String appId, @Query("salt") int salt, @Query("sign") String sign);

}

File: app/src/main/java/com/github/pwcong/jpstart/network/baidu/api/BaiduTranslateApi.kt
package com.github.pwcong.jpstart.network.baidu.api

import com.github.pwcong.jpstart.mvp.bean.BaiduTranslateBean
import retrofit2.http.GET
import retrofit2.http.Query
import rx.Observable

interface BaiduTranslateApi {
    @GET("/api/trans/vip/translate")
    fun request(
        @Query("q") q: String?, @Query("from") from: String?, @Query("to") to: String?,
        @Query("appid") appId: String?, @Query("salt") salt: Int, @Query("sign") sign: String?
    ): Observable<BaiduTranslateBean>

    companion object {
        const val BAIDU_TRANSLATE_URL: String = "http://api.fanyi.baidu.com"
        const val BAIDU_TRANSLATE_APPID: String = "******"
        const val BAIDU_TRANSLATE_SECRETKEY: String = "******"

        const val AUTO: String = "auto"
        const val ZH: String = "zh"
        const val JP: String = "jp"
        const val EN: String = "en"
    }
}
